tests/RouteTest.php

```diff
@@ -169,46 +170,155 @@ static function (): ResponseInterface {
         $this->assertSame(418, $response->getStatusCode());
     }
 
-    public function testThen(): void
+    public function testThenByStatusCode(): void
```

Comment:
There are two tests in this test-method:

1. Checking that whole middleware stack for a route is processed.
2. Checking that middleware stack execution could be interrupted.

Worth splitting these into two separate test methods.

Reply:
I agree, and have a question https://github.com/yiisoft/router/pull/20#discussion_r373853660

Reply:
Done.